scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -4225,28 +4219,23 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
     )
 
     collectUparams()
-
     newLineOptWhenFollowedBy[LeftBrace]
 
-    val methodsAll: List[Stat] =
+    val body: Stat =
       if (token.is[LeftBrace]) {
-        inBraces(templateStats())
+        autoPos(Term.Block(inBraces(templateStats())))
       } else {
         newLinesOpt()
         if (in.observeIndented()) {
-          indented(templateStats())
+          val block = autoPos(Term.Block(indented(templateStats())))
+          if (block.stats.size == 1) block.stats.head
```

Comment:
does this mean you will NOT keep braces for this single-stat block? i would prefer that you had. if block had braces, they need to be part of it, regardless of the number of stats.

Reply:
That's only for indentation, that doesn't have braces at all.

Reply:
zrozumiał...